src/libraries/System.Private.CoreLib/src/System/Threading/PreAllocatedOverlapped.cs

```diff
@@ -52,69 +53,51 @@ public sealed class PreAllocatedOverlapped : IDisposable, IDeferredDisposable
-        ///     a delegate that is invoked when each asynchronous I/O operation is complete, a user-provided
-        ///     object providing context, and managed objects that serve as buffers.
-        /// </summary>
-        /// <param name="callback">
-        ///     An <see cref="IOCompletionCallback"/> delegate that represents the callback method
-        ///     invoked when each asynchronous I/O operation completes.
-        /// </param>
-        /// <param name="state">
-        ///     A user-provided object that distinguishes <see cref="NativeOverlapped"/> instance produced from this
-        ///     object from other <see cref="NativeOverlapped"/> instances. Can be <see langword="null"/>.
-        /// </param>
-        /// <param name="pinData">
-        ///     An object or array of objects representing the input or output buffer for the operations. Each
-        ///     object represents a buffer, for example an array of bytes.  Can be <see langword="null"/>.
-        /// </param>
-        /// <remarks>
-        ///     The new <see cref="PreAllocatedOverlapped"/> instance can be passed to
-        ///     <see cref="ThreadPoolBoundHandle.AllocateNativeOverlapped(PreAllocatedOverlapped)"/>, to produce
-        ///     a <see cref="NativeOverlapped"/> instance that can be passed to the operating system in overlapped
-        ///     I/O operations.  A single <see cref="PreAllocatedOverlapped"/> instance can only be used for
-        ///     a single native I/O operation at a time.  However, the state stored in the <see cref="PreAllocatedOverlapped"/>
-        ///     instance can be reused for subsequent native operations. ExecutionContext is not flowed to the invocation
-        ///     of the callback.
-        ///     <note>
-        ///         The buffers specified in <paramref name="pinData"/> are pinned until <see cref="Dispose"/> is called.
-        ///     </note>
-        /// </remarks>
-        /// <exception cref="ArgumentNullException">
-        ///     <paramref name="callback"/> is <see langword="null"/>.
-        /// </exception>
-        /// <exception cref="ObjectDisposedException">
-        ///     This method was called after the <see cref="ThreadPoolBoundHandle"/> was disposed.
-        /// </exception>
         [CLSCompliant(false)]
         public static PreAllocatedOverlapped UnsafeCreate(IOCompletionCallback callback, object? state, object? pinData) =>
-            new PreAllocatedOverlapped(callback, state, pinData, flowExecutionContext: false);
+            ThreadPool.UseWindowsThreadPool ? UnsafeCreateCore(callback, state, pinData) : UnsafeCreatePortableCore(callback, state, pinData);
 
-        private PreAllocatedOverlapped(IOCompletionCallback callback, object? state, object? pinData, bool flowExecutionContext)
+        private unsafe PreAllocatedOverlapped(IOCompletionCallback callback, object? state, object? pinData, bool flowExecutionContext)
```

Comment:
I think so, the assignment of `_overlappedCore`  raises `error CS0214: Pointers and fixed size buffers may only be used in an unsafe context` otherwise